app/src/Controller/MessageController.php

```diff
@@ -144,19 +145,21 @@ public function index(Request $request, PaginatorInterface $paginator, ?string $
         }
         $filterForm->get('per_page')->setData($per_page);
 
-        $msg2Show = $paginator->paginate(
-            $allMessages,
-            $request->query->getInt('page', 1)/* page number */,
-            $per_page
+        $messagesRecipients = $paginator->paginate(
+            $messagesRecipientsQuery,
+            $request->query->getInt('page', 1),
+            $per_page,
+            [
+                'wrap-queries' => true,
+                'fetchJoinCollection' => false,
+                'distinct' => false,
+            ]
         );
-        unset($allMessages);
+        // dd($messagesRecipients);
         return $this->render('message/index.html.twig', [
-            'controller_name' => 'MessageController',
             'subTitle' => $subTitle,
-            'msgs' => $msg2Show,
-            'type' => $type,
-            'per_page_global' => $per_page,
-            'totalItemFound' => $totalMsgFound,
+            'messagesRecipients' => $messagesRecipients,
+            'type' => $messageStatus?->getId(),
```

Comment:
Est-ce qu’on ne renommerait pas cette variable par `messageStatus` pour plus de cohérence ? Si oui, il faudra aussi changer dans le template 

```suggestion
            'messageStatus' => $messageStatus,
```

Reply:
oui, je suis d'accord. Je m'occupe du template